Content.Shared/_Harmony/ItemCurse/SharedItemCurseSystem.cs

```diff
@@ -0,0 +1,239 @@
+            //
+            // For some reason client thinks the station grid owns the action on client and this doesn't work. It doesn't work in PopupEntity(mispredicts) and PopupPredicted either(doesnt show).
+            // I don't have the heart to move this code to server because of this small thing.
+            // This line will only do something once that is fixed.
+            if (instantAction.AttachedEntity != null)
+            {
+                _popups.PopupClient(Loc.GetString("item-recall-item-unmark", ("item", item)), instantAction.AttachedEntity.Value, instantAction.AttachedEntity.Value, PopupType.MediumCaution);
+                RemoveFromPvsOverride(item, instantAction.AttachedEntity.Value);
+            }
+
+            action.MarkedEntity = null;
+            UpdateActionAppearance((marker.MarkedByAction.Value, action));
+            Dirty(marker.MarkedByAction.Value, action);
+        }
+
+        RemCompDeferred<CurseMarkerComponent>(item);
+    }
+
+    private void UpdateActionAppearance(Entity<ItemCurseComponent> action)
+    {
+        if (!TryComp<InstantActionComponent>(action, out var instantAction))
+            return;
+
+        if (action.Comp.MarkedEntity == null)
+        {
+            if (action.Comp.InitialName != null)
+                _metaData.SetEntityName(action, action.Comp.InitialName);
+            if (action.Comp.InitialDescription != null)
+                _metaData.SetEntityDescription(action, action.Comp.InitialDescription);
+            _actions.SetEntityIcon(action, null, instantAction);
+        }
+        else
+        {
+            if (action.Comp.WhileMarkedName != null)
+                _metaData.SetEntityName(action, Loc.GetString(action.Comp.WhileMarkedName,
+                    ("item", action.Comp.MarkedEntity.Value)));
+
+            if (action.Comp.WhileMarkedDescription != null)
+                _metaData.SetEntityDescription(action, Loc.GetString(action.Comp.WhileMarkedDescription,
+                    ("item", action.Comp.MarkedEntity.Value)));
+
+            _actions.SetEntityIcon(action, action.Comp.MarkedEntity, instantAction);
+        }
+    }
+
+    private void AddToPvsOverride(EntityUid uid, EntityUid user)
+    {
+        if (!_player.TryGetSessionByEntity(user, out var mindSession))
+            return;
+
+        _pvs.AddSessionOverride(uid, mindSession);
+    }
+
+    private void RemoveFromPvsOverride(EntityUid uid, EntityUid user)
+    {
+        if (!_player.TryGetSessionByEntity(user, out var mindSession))
+            return;
+
+        _pvs.RemoveSessionOverride(uid, mindSession);
+    }
+
+    // Methods to be overridden in server ItemCurseSystem
+    public virtual void CreateLightning(EntityUid ent, ItemCurseComponent comp)
+    {
+
+    }
+
+    public virtual void ShockHolder(EntityUid ent, EntityUid source, ItemCurseComponent comp)
+    {
+
+    }
+
+    public virtual void Snap(EntityUid ent)
+    {
+
+    }
```

Comment:
These functions don't need to be public
```suggestion
    protected virtual void CreateLightning(EntityUid ent, ItemCurseComponent comp)
    {

    }

    protected virtual void ShockHolder(EntityUid ent, EntityUid source, ItemCurseComponent comp)
    {

    }

    protected virtual void Snap(EntityUid ent)
    {

    }
```